File: app/src/main/java/com/dimatest/movieapp/repositories/remote/MovieRemoteRepository.java
package com.dimatest.movieapp.repositories.remote;

import com.dimatest.movieapp.BuildConfig;
import com.dimatest.movieapp.network.MovieService;
import com.dimatest.movieapp.network.entity.MovieResponse;

import javax.inject.Inject;

import io.reactivex.Single;

public class MovieRemoteRepository implements MovieRemoteRepositoryInterface {

    private MovieService movieService;

    @Inject
    public MovieRemoteRepository(MovieService movieService) {
        this.movieService = movieService;
    }

    @Override
    public Single<MovieResponse> getMovies(int page) {
        return movieService.getMovies(BuildConfig.API_KEY, page);
    }
}

File: app/src/main/java/com/dimatest/movieapp/repositories/remote/MovieRemoteRepository.kt
package com.dimatest.movieapp.repositories.remote

import com.dimatest.movieapp.BuildConfig
import com.dimatest.movieapp.network.MovieService
import com.dimatest.movieapp.network.entity.MovieResponse
import io.reactivex.Single

class MovieRemoteRepository (private val movieService: MovieService) : MovieRemoteRepositoryInterface {

    override fun getMovies(page: Int): Single<MovieResponse> {
        return movieService.getMovies(BuildConfig.API_KEY, page)
    }
}